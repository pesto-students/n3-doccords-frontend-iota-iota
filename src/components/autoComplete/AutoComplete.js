import React from "react";
import TextField from "@material-ui/core/TextField";
import Autocomplete from "@material-ui/lab/Autocomplete";

// eslint-disable-next-line react/prop-types
export default function MyAutoComplete({ list, handleHealthSelection, value }) {
  return (
    <Autocomplete
      id="combo-box-demo"
      // eslint-disable-next-line react/prop-types
      options={list.map((t) => t)}
      getOptionLabel={(option) => option.title}
      style={{ width: 400 }}
      value={value}
      onChange={(event, value) =>
        value !== null ? handleHealthSelection(value) : console.log("its null")
      }
      renderInput={(params) => (
        <TextField {...params} label="Select Health Topic" variant="outlined" />
      )}
    />
  );
}
