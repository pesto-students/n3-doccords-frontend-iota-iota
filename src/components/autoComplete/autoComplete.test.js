import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import AutoComplete from "components/autoComplete/AutoComplete";

describe("AutoComplete", () => {
  const list = [
    {
      title: "Item 1",
    },
    {
      title: "Item 2",
    },
  ];

  it("rendering without Error", () => {
    const wrapper = shallow(
      <AutoComplete list={list} handleHealthSelection={() => {}} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
