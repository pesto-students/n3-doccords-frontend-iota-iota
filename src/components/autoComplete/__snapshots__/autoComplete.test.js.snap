// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoComplete rendering without Error 1`] = `
<WithStyles(ForwardRef(Autocomplete))
  getOptionLabel={[Function]}
  id="combo-box-demo"
  onChange={[Function]}
  options={
    Array [
      Object {
        "title": "Item 1",
      },
      Object {
        "title": "Item 2",
      },
    ]
  }
  renderInput={[Function]}
  style={
    Object {
      "width": 400,
    }
  }
/>
`;
